extensions/authentication-jwt/README.md

```diff
@@ -0,0 +1,209 @@
+  ) {}
+
+  async handle(context: RequestContext) {
+    try {
+      const {request, response} = context;
+      const route = this.findRoute(request);
+      // - enable jwt auth -
+      // call authentication action
+      await this.authenticateRequest(request);
+
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+      this.send(response, result);
+    } catch (error) {
+      // - enable jwt auth -
+      // improve the error check
+      if (
+        error.code === AUTHENTICATION_STRATEGY_NOT_FOUND ||
+        error.code === USER_PROFILE_NOT_FOUND
+      ) {
+        Object.assign(error, {statusCode: 401 /* Unauthorized */});
+      }
+      this.reject(context, error);
+    }
+  }
+}
+```
+
+</p>
+</details>
+
+- mount jwt component in application
+
+<details>
+<summary><strong>Check The Code</strong></summary>
+<p>
+
+```ts
+export class TestApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    // Set up the custom sequence
+    this.sequence(MySequence);
+
+    // Set up default home page
+    this.static('/', path.join(__dirname, '../public'));
+
+    // - enable jwt auth -
+    // Add security spec (Future work: refactor it to an enhancer)
+    this.addSecuritySpec();
+    // Mount authentication system
+    this.component(AuthenticationComponent);
+    // Mount jwt component
+    this.component(JWTAuthenticationComponent);
+    // Bind datasource
+    this.dataSource(DbDataSource, UserServiceBindings.DATASOURCE_NAME);
+
+    this.component(RestExplorerComponent);
+    this.projectRoot = __dirname;
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {};
+  }
+
+  // - enable jwt auth -
+  // Currently there is an extra function to
+  // merge the security spec into the application.
+  // This will be improved with a coming enhancer.
+  // See section [Future Work](#future-work)
+  addSecuritySpec(): void {
+    this.api({
+      openapi: '3.0.0',
+      info: {
+        title: 'test application',
+        version: '1.0.0',
+      },
+      paths: {},
+      components: {securitySchemes: SECURITY_SCHEME_SPEC},
+      security: [
+        {
+          // secure all endpoints with 'jwt'
+          jwt: [],
+        },
+      ],
+      servers: [{url: '/'}],
+    });
+  }
+}
+```
+
+</p>
+</details>
+
+_All the jwt authentication related code are marked with comment "- enable jwt
+auth -", you can search for it to find all the related code you need to enable
+the entire jwt authentication in a LoopBack 4 application._
+
+## Add Endpoint in Controller
+
+After mounting the component, you can call token and user services to perform
+login then decorate endpoints with `@authentication('jwt')` to inject the logged
+in user's profile.
+
+This module contains an example LoopBack 4 application in folder `fixtures`. It
```

Comment:
"This module contains an example LoopBack 4 application in folder `fixtures`." -> "This module contains an example application in the `fixtures` folder."?

The term "LoopBack 4 " is redundant in the context of the extension.